### ISSUE TITLE
Accessibility: 'Clear all' button should be triggered on mouseup

### ISSUE THREAD

**Issue by andreasnyberg** [ng-select/ng-select#2661]
### Is your feature request related to a problem? Please describe

The 'Clear all' button click is currently triggered on mouse**down**. It can cause "accidental activation of controls by mouse or touch" [according to WCAG](https://www.w3.org/WAI/WCAG21/Understanding/pointer-cancellation.html).

### Describe the solution you'd like

Should be triggered on mouse**up**: https://www.w3.org/TR/WCAG21/#pointer-cancellation

### Describe alternatives you've considered

.

### Additional context

<img width="500" height="129" alt="Image" src="https://github.com/user-attachments/assets/1542d352-e3d4-410a-86a7-248dc6490a5e" />
